File: app/src/main/java/com/wxm/KeepAccount/Base/db/UsrItem.java
package com.wxm.KeepAccount.Base.db;

import com.j256.ormlite.field.DataType;
import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

/**
 * 用户数据
 * Created by 123 on 2016/8/5.
 */
@DatabaseTable(tableName = "tbUsr")
public class UsrItem {
    public final static String FIELD_ID = "_id";
    public final static String FIELD_NAME = "name";
    public final static String FIELD_PWD = "pwd";

    @DatabaseField(generatedId = true, columnName = "_id", dataType = DataType.INTEGER)
    private int _id;

    @DatabaseField(columnName = "name", unique = true, dataType = DataType.STRING)
    private String name;

    @DatabaseField(columnName = "pwd", dataType = DataType.STRING)
    private String pwd;

    public UsrItem()    {
        setName("");
        setPwd("");
        setId(-1);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPwd() {
        return pwd;
    }

    public void setPwd(String pwd) {
        this.pwd = pwd;
    }

    public int getId() {
        return _id;
    }

    public void setId(int _id) {
        this._id = _id;
    }
}

File: app/src/main/java/wxm/KeepAccount/item/UsrItem.kt
package wxm.KeepAccount.item

import com.j256.ormlite.field.DataType
import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable
import wxm.KeepAccount.define.GlobalDef
import wxm.KeepAccount.define.IPublicClone
import wxm.KeepAccount.utility.defaultUsrIcon
import wxm.androidutil.db.IDBRow


/**
 * usr class
 * Created by WangXM on 2016/8/5.
 */
@DatabaseTable(tableName = "tbUsr")
class UsrItem : IDBRow<Int>, Cloneable, IPublicClone {
    @DatabaseField(generatedId = true, columnName = FIELD_ID, dataType = DataType.INTEGER)
    var id: Int = GlobalDef.INVALID_ID

    @DatabaseField(columnName = FIELD_NAME, unique = true, dataType = DataType.STRING)
    var name: String = ""

    @DatabaseField(columnName = FIELD_PWD, dataType = DataType.STRING)
    var pwd: String = ""

    @DatabaseField(columnName = FIELD_ICON_PATH, dataType = DataType.STRING)
    var iconPath: String = defaultUsrIcon()

    override fun getID(): Int {
        return id
    }

    override fun setID(integer: Int) {
        id = integer
    }

    override fun clone(): Any {
        return UsrItem().let{
            it.id = this.id
            it.name = this.name
            it.pwd = this.pwd
            it.iconPath = this.iconPath

            it
        }
    }

    override fun publicClone(): Any {
        return clone()
    }

    companion object {
        const val FIELD_ID = "_id"
        const val FIELD_NAME = "name"
        const val FIELD_PWD = "pwd"
        const val FIELD_ICON_PATH = "icon_path"
    }
}
